main.go

```diff
@@ -35,13 +42,264 @@ func main() {
 //	number of line deleted
 //	list of function calls seen in the diffs and their number of calls
 func computeDiff() *diffResult {
+	// Index all files in the diffs folder
+	rootFolder := "./diffs/"
+	diffFiles, err := ioutil.ReadDir(rootFolder)
+	if err != nil {
+		log.Fatal(err)
+	}
 
-	return nil
+	// Start a routine for each file
+	resChannel := make(chan diffResult)
+	expectedResults := 0
+	for _, file := range diffFiles {
+		go parseFileDiff(rootFolder+file.Name(), resChannel)
+		expectedResults++
+	}
+
+	diffResult := diffResult{
+		functionCalls: make(map[string]int),
+	}
+
+	// Get results from the communication channel
+	for i := 0; i < expectedResults; i++ {
+		routineRes := <-resChannel
+
+		diffResult.regions += routineRes.regions
+		diffResult.lineAdded += routineRes.lineAdded
+		diffResult.lineDeleted += routineRes.lineDeleted
+		diffResult.files = append(diffResult.files, routineRes.files...)
+
+		for key, value := range routineRes.functionCalls {
+			diffResult.functionCalls[key] += value
+		}
+	}
+
+	return &diffResult
+}
+
+// Parse the content of a diff file and send the result through the channel
+func parseFileDiff(filename string, resChannel chan diffResult) {
+	// Define all the regex that we need to parse the Diffs
+	var regionPrefix = "^@@"
+	var diffCallPrefix = "diff --git "
+	var diffCallRegex = regexp.MustCompile(regexp.QuoteMeta(diffCallPrefix))
+	var deletedPrefix = "^" + regexp.QuoteMeta("- ")
```

Comment:
Generally `regexp.QuoteMeta` can be used to dynamically unquote regexes computed at runtime, for static stuff, I'm not sure of how useful that is, you're probably better off quoting what's problematic. Besides, safe for the `+` one, none actually requires any kind of quoting